lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -19,4 +20,20 @@ def field_dictionary(self, **kwargs):
             user_enrollments = CourseEnrollment.enrollments_for_user(kwargs['user'])
             field_dictionary['course'] = [unicode(enrollment.course_id) for enrollment in user_enrollments]
 
+        # if we have an org filter, ony include results for this org filter
+        course_org_filter = microsite.get_value('course_org_filter')
+        if course_org_filter:
+            field_dictionary['org'] = course_org_filter
+
         return field_dictionary
+
+    def exclude_dictionary(self, **kwargs):
```

Comment:
Ah, that's where the search app comes in...

The field_, filter_ and exclude_ dictionaries are how the LMS injects it's rules into the search procedure. This is done through dependency injection where the `LMSSearchFilterGenerator` (which overrides the default `SearchFilterGenerator`) gets loaded and used to provide LMS-specific filters for the search.